.test-infra/jenkins/verify_performance_test_results.py

```diff
@@ -0,0 +1,259 @@
+        result_list = []
+        for row in results:
+            result_list.append(row.query_result)
+        return result_list
+    elif return_type == "value":
+        # All queries must have single element in output displayed as query_result
+        for row in results:
+            return (row.query_result)
+    else:
+        print("This type is not supported")
+        return None
+
+def count_queries(table_name, time_start, time_stop, metric):
+    # This function checks how many data was inserted in time interval.
+    sql_command = 'select count(*) as query_result from {} where TIMESTAMP > {} and TIMESTAMP < {} and METRIC=\'{}\''.format(
+        table_name,
+        time_start,
+        time_stop,
+        metric
+        )
+    count = submit_big_query_job(sql_command, "value")
+    print("Number of records inserted into {} between {} - {}: {}".format(
+        table_name,
+        time.strftime(TIME_PATTERN, time.gmtime(time_start)),
+        time.strftime(TIME_PATTERN, time.gmtime(time_stop)),
+        count))
+    return count
+
+def get_average_from(table_name, time_start, time_stop, metric):
+    # This function return average value of the provided metric in time interval.
+    sql_command = 'select avg(value) as query_result from {} where TIMESTAMP > {} and TIMESTAMP < {} and METRIC=\'{}\''.format(
+        table_name,
+        time_start,
+        time_stop,
+        metric
+    )
+    average = submit_big_query_job(sql_command, "value")
+    return average
+
+def get_stddev_from(table_name, time_start, time_stop, metric):
+    # This function return standard deviation of the provided metric in time interval.
+    sql_command = 'select stddev(value) as query_result from {} where TIMESTAMP > {} and TIMESTAMP < {} and METRIC=\'{}\''.format(
+        table_name,
+        time_start,
+        time_stop,
+        metric
+    )
+    stddev = submit_big_query_job(sql_command, "value")
+    return stddev
+
+def get_records_from(table_name, time_start, time_stop, metric, number_of_records):
+    # This function checks how many data was inserted in time interval.
+    sql_command = 'select value as query_result from {} where TIMESTAMP > {} and TIMESTAMP < {} and METRIC=\'{}\' ORDER BY TIMESTAMP DESC LIMIT {}'.format(
+        table_name,
+        time_start,
+        time_stop,
+        metric,
+        number_of_records
+        )
+    print(sql_command)
+    list_of_records = submit_big_query_job(sql_command, "list")
+    return list_of_records
+
+def create_report(bqtables):
+    # This function create a report message from tables.
+    report_message = ''
+    slack_report_message = ''
+
+    for bqtable in bqtables:
+        # Get raw table name
+        bq_table_name = re.sub(r'\"|\[|\]', '', bqtable).strip()
+
+        # Make sure there was data records inserted
+        nb_recent_records = count_queries(bq_table_name, TIME_POINT_1, NOW, metric)
+        if nb_recent_records == 0:
```

Comment:
Maybe this comment is unclear. The variable nb_recent_records was introduced for this purpose and is storing result of this test by executing COUNT on BigQuery table and checking how many data was uploaded recently., or do you mean writing additional python testing method?

Reply:
Sorry, what I meant was that we should have unit test coverage for anomaly detection logic. I think we should make sure that this code is stable since we want people to trust and act on signals provided through this logic.